transaction_test.go

```diff
@@ -27,6 +27,9 @@ func maybeEndTransaction(t *testing.T, tx *Transaction) {
 
 func ensureTransactionEnds(t *testing.T, tx *Transaction) {
 	t.Helper()
+	if tx == nil {
+		return
+	}
```

Comment:
More than doing this, it's probably better to avoid to call this at all when tx is null, so that we fail in case it's unexpected, so I'd fix the caller order instead.

Reply:
I think I got all of these, would appreciate a second look.